File: Server/ChatServer/src/main/java/com/szlazakm/chatserver/dtos/SPKCreateDTO.java
package com.szlazakm.chatserver.dtos;


import lombok.Builder;
import lombok.Value;
import lombok.extern.jackson.Jacksonized;

import java.util.UUID;

@Value
@Builder
@Jacksonized
public class SPKCreateDTO {

    String phoneNumber;
    int id;
    String signedPreKey;
    String signature;
    Long timestamp;
}

File: client/app/src/main/java/com/szlazakm/safechat/webclient/dtos/SPKCreateDTO.kt
package com.szlazakm.safechat.webclient.dtos

import com.fasterxml.jackson.annotation.JsonProperty

data class SPKCreateDTO (
    @JsonProperty("phoneNumber")
    val phoneNumber: String,
    @JsonProperty("id")
    val id: Int,
    @JsonProperty("signedPreKey")
    val signedPreKey: String,
    @JsonProperty("signature")
    val signature: String,
    @JsonProperty("timestamp")
    val timestamp: Long
)
